### PR TITLE
Add paper trail to document tag and annotations

### PR DESCRIPTION
**Pull request by mdbenjam**
Connects: https://github.com/department-of-veterans-affairs/caseflow/issues/2546

We added paper trails to document tags and annotations. We had to add a primary key to the document tags table since it was created as a join table without one.

**Test Plan**
- [ ] Run migration locally and ensure creation and destruction of tags and annotations is versioned in paper trail.
- [ ] Verify changes to annotations are captured in paper trail.

### CODE REVIEW

File: app/controllers/document_controller.rb [hunk 1]
```diff
@@ -2,7 +2,6 @@
```
Comment (dannysellers) [inline app/controllers/document_controller.rb]: Nit: this import can probably be removed

Comment (mdbenjam) [reply app/controllers/document_controller.rb]: Sorry import? You mean space? Definitely. I'll remove it :)

Comment (dannysellers) [reply app/controllers/document_controller.rb]: No, sorry, I meant the `require` line above this. GH wouldn't let me make a comment on that line cause I guess it was too far away from code you'd modified. I only added it in the first place cause RubyMine threw a weak and strangely-worded warning about not being able to find `:set_paper_trail_whodunnit`, but it doesn't seem to matter functionally..

Comment (mdbenjam) [reply app/controllers/document_controller.rb]: ah oops. I'll remove it on another PR.

**Pull request merged**